From e601d836419d757fc2801fd4b17d7cae6eec724d Mon Sep 17 00:00:00 2001
From: Mike Weber <mike@weberapps.com>
Date: Mon, 29 Sep 2008 15:38:41 -0500
Subject: [PATCH] * Added the migration version to the file contents to make re-migrating certain migration versions easier for developers.
 * Updated the documentation for the migration method to reflect the new timestamp style for migration filenames
 * Fixed a spelling mistake in the ModelGenerator class

---
 activerecord/lib/active_record/migration.rb        |    3 ++-
 .../components/migration/templates/migration.rb    |    1 +
 .../generators/components/model/model_generator.rb |    2 +-
 .../components/model/templates/migration.rb        |    1 +
 4 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/activerecord/lib/active_record/migration.rb b/activerecord/lib/active_record/migration.rb
index 1d843ff..4c4e7b5 100644
--- a/activerecord/lib/active_record/migration.rb
+++ b/activerecord/lib/active_record/migration.rb
@@ -108,7 +108,8 @@ module ActiveRecord
   #
   # where MyNewMigration is the name of your migration. The generator will
   # create an empty migration file <tt>nnn_my_new_migration.rb</tt> in the <tt>db/migrate/</tt>
-  # directory where <tt>nnn</tt> is the next largest migration number.
+  # directory where <tt>nnn</tt> is a timestamp or, if ActiveRecord::Base.timestamped_migrations
+  # is set to false, the next largest migration number.
   #
   # You may then edit the <tt>self.up</tt> and <tt>self.down</tt> methods of
   # MyNewMigration.
diff --git a/railties/lib/rails_generator/generators/components/migration/templates/migration.rb b/railties/lib/rails_generator/generators/components/migration/templates/migration.rb
index ca35a43..83740c9 100644
--- a/railties/lib/rails_generator/generators/components/migration/templates/migration.rb
+++ b/railties/lib/rails_generator/generators/components/migration/templates/migration.rb
@@ -1,3 +1,4 @@
+# VERSION=<%= current_migration_number %>
 class <%= class_name.underscore.camelize %> < ActiveRecord::Migration
   def self.up<% attributes.each do |attribute| %>
     <%= migration_action %>_column :<%= table_name %>, :<%= attribute.name %><% if migration_action == 'add' %>, :<%= attribute.type %><% end -%>
diff --git a/railties/lib/rails_generator/generators/components/model/model_generator.rb b/railties/lib/rails_generator/generators/components/model/model_generator.rb
index 9be9cad..a974ffd 100644
--- a/railties/lib/rails_generator/generators/components/model/model_generator.rb
+++ b/railties/lib/rails_generator/generators/components/model/model_generator.rb
@@ -40,6 +40,6 @@ class ModelGenerator < Rails::Generator::NamedBase
       opt.on("--skip-migration", 
              "Don't generate a migration file for this model") { |v| options[:skip_migration] = v }
       opt.on("--skip-fixture",
-             "Don't generation a fixture file for this model") { |v| options[:skip_fixture] = v}
+             "Don't generate a fixture file for this model") { |v| options[:skip_fixture] = v}
     end
 end
diff --git a/railties/lib/rails_generator/generators/components/model/templates/migration.rb b/railties/lib/rails_generator/generators/components/model/templates/migration.rb
index 382fd11..af01e03 100644
--- a/railties/lib/rails_generator/generators/components/model/templates/migration.rb
+++ b/railties/lib/rails_generator/generators/components/model/templates/migration.rb
@@ -1,3 +1,4 @@
+# VERSION=<%= current_migration_number %>
 class <%= migration_name %> < ActiveRecord::Migration
   def self.up
     create_table :<%= table_name %> do |t|
-- 
1.5.6.3

